src/stdatamodels/jwst/datamodels/wcs_ref_models.py

```diff
@@ -459,6 +460,83 @@ def to_fits(self):
         raise NotImplementedError("FITS format is not supported for this file.")
 
 
+class MiriWFSSSpecwcsModel(ReferenceFileModel):
+    """A model for a reference file of type "specwcs" for MIRI WFSS."""
+
+    schema_url = "http://stsci.edu/schemas/jwst_datamodel/specwcs_miri_wfss.schema"
+    reftype = "specwcs"
+
+    def __init__(
+        self,
+        init=None,
+        displ=None,
+        dispx=None,
+        dispy=None,
+        invdispl=None,
+        orders=None,
+        **kwargs,
+    ):
+        """
+        Initialize the model.
+
+        Parameters
+        ----------
+        init : str, None, optional
+            File name for input file in ASDF format from which to initialize the model.
+        displ : list of `~astropy.modeling.Model`
+            MIRI WFSS wavelength dispersion model
+        dispx : list of `~astropy.modeling.Model`
+            MIRI WFSS row dispersion model
+        dispy : list of `~astropy.modeling.Model`
+            MIRI WFSS column dispersion model
+        invdispl : list of `~astropy.modeling.Model`
+            MIRI WFSS inverse wavelength dispersion model
+        orders : list of int
+            MIRI WFSS orders, matched to the array locations of the
+            dispersion models
+        **kwargs
+            Additional keyword arguments to pass to ReferenceFileModel
+        """
+        super().__init__(init=init, **kwargs)
+
+        if init is None:
+            self.populate_meta()
+        if displ is not None:
+            self.displ = displ
+        if dispx is not None:
+            self.dispx = dispx
+        if dispy is not None:
+            self.dispy = dispy
+        if invdispl is not None:
+            self.invdispl = invdispl
+        if orders is not None:
+            self.orders = orders
+
+    def populate_meta(self):
+        self.meta.instrument.name = "MIRI"
+        self.meta.instrument.detector = "MIRIMAGE"
+        self.meta.exposure.type = "MIR_WFSS"
+        self.meta.reftype = self.reftype
+
+    def validate(self):
+        super(MiriWFSSSpecwcsModel, self).validate()
+        try:
+            assert isinstance(self.meta.input_units, (str, u.NamedUnit))
+            assert isinstance(self.meta.output_units, (str, u.NamedUnit))
+            assert self.meta.instrument.name == "MIRI"
+            assert self.meta.exposure.type == "MIR_WFSS"
+            assert self.meta.reftype == self.reftype
+            assert len(self.orders) == 1
```

Comment:
```suggestion
            n_orders = len(self.orders)
            assert n_order == 1
            assert len(self.displ) == n_orders
            assert len(self.dispx) == n_orders
            assert len(self.dispy) == n_orders
            assert all((len(sub_y) == 3 for sub_y in self.dispy))
```
Since the code has some expectations that aren't in the schema, how about adding these additional checks here?

Reply:
@braingram do these look like a good idea to also add for NIRISS and NIRCam?  If so, would you mind adding a comment to https://github.com/spacetelescope/stdatamodels/issues/542 ?